test/rulesets/Base/indexing.jl

```diff
@@ -48,3 +48,32 @@
         end
     end
 end
+
+if VERSION >= v"1.6" # Before Julia 1.6, `collect∘eachrow` is not a specific type.
+
+@testset "eachslice" begin
+    # Testing eachrow not collect∘eachrow leads to errors, e.g.
+    # test_rrule: eachrow on Vector{Float64}: Error During Test at /Users/me/.julia/packages/ChainRulesTestUtils/8dFTY/src/testers.jl:195
+    #   Got exception outside of a @test
+    #   DimensionMismatch("second dimension of A, 6, does not match length of x, 5")
+
+    test_rrule(collect∘eachrow, rand(5); check_inferred=false)
+    test_rrule(collect∘eachrow, rand(3, 4); check_inferred=false)
+
+    test_rrule(collect∘eachcol, rand(3, 4); check_inferred=false)
+    test_rrule(collect∘eachcol, Diagonal(rand(5)); check_inferred=false)
```

Comment:
This passes locally, and on nightly, but fails on 1.7 with:
```
Testing rulesets/Base/indexing.jl:
117
test_rrule: ComposedFunction{typeof(collect), typeof(eachcol)}(collect, eachcol) on Diagonal{Float64, Vector{Float64}}: Error During Test at /home/runner/.julia/packages/ChainRulesTestUtils/8dFTY/src/testers.jl:195
118
  Got exception outside of a @test
119
  DimensionMismatch("second dimension of A, 15, does not match length of x, 25")
120
  Stacktrace:
121
    [1] gemv!(y::Vector{Float64}, tA::Char, A::Matrix{Float64}, x::Vector{Float64}, α::Bool, β::Bool)
122
      @ LinearAlgebra /opt/hostedtoolcache/julia/1.7.1/x64/share/julia/stdlib/v1.7/LinearAlgebra/src/matmul.jl:530
123
    [2] mul!
124
      @ /opt/hostedtoolcache/julia/1.7.1/x64/share/julia/stdlib/v1.7/LinearAlgebra/src/matmul.jl:97 [inlined]
125
    [3] mul!
126
      @ /opt/hostedtoolcache/julia/1.7.1/x64/share/julia/stdlib/v1.7/LinearAlgebra/src/matmul.jl:275 [inlined]
127
    [4] *(transA::Transpose{Float64, Matrix{Float64}}, x::Vector{Float64})
128
      @ LinearAlgebra /opt/hostedtoolcache/julia/1.7.1/x64/share/julia/stdlib/v1.7/LinearAlgebra/src/matmul.jl:87
129
    [5] _j′vp(fdm::FiniteDifferences.AdaptedFiniteDifferenceMethod{5, 1, FiniteDifferences.UnadaptedFiniteDifferenceMethod{7, 5}}, f::Function, ȳ::Vector{Float64}, x::Vector{Float64})
130
      @ FiniteDifferences ~/.julia/packages/FiniteDifferences/iFBBx/src/grad.jl:80
131
    [6] j′vp(fdm::FiniteDifferences.AdaptedFiniteDifferenceMethod{5, 1, FiniteDifferences.UnadaptedFiniteDifferenceMethod{7, 5}}, f::Function, ȳ::Vector{Vector{Float64}}, x::Diagonal{Float64, Vector{Float64}})
132
```
Same versions locally:
```
(jl_1Is8F5) pkg> st
Status `/private/var/folders/yq/4p2zwd614y59gszh7y9ypyhh0000gn/T/jl_1Is8F5/Project.toml`
  [082447d4] ChainRules v1.21.0 `~/.julia/dev/ChainRules`
  [cdddcdb0] ChainRulesTestUtils v1.3.1
  [26cc04aa] FiniteDifferences v0.12.20
  ```